src/layer-composer/generators/track/track.ts

```diff
@@ -1,16 +1,67 @@
-import { GeneratorConfig } from 'layer-composer/types'
-// TODO custom "augemented" GeoJSON type
+import { GeneratorConfig, Dictionary } from 'layer-composer/types'
+import memoizeOne from 'memoize-one'
+import { scaleLinear, scalePow } from 'd3-scale'
+// TODO custom "augmented" GeoJSON type?
 // see https://github.com/yagajs/generic-geojson/blob/master/index.d.ts
-import { FeatureCollection } from 'geojson'
+import { FeatureCollection, LineString } from 'geojson'
 import filterGeoJSONByTimerange from './filterGeoJSONByTimerange'
+import { simplifyTrack } from './simplify-track'
 
 export const TRACK_TYPE = 'TRACK'
 
+const mapZoomToMinPosΔ = (zoomLoadLevel: number) => {
+  // first normalize and invert z level
+  const normalizedZoom = scaleLinear()
+    .clamp(true)
+    .range([1, 0])
+    .domain([3, 12])(zoomLoadLevel)
+
+  const MIN_POS_Δ_LOW_ZOOM = 0.1
+  const MIN_POS_Δ_HIGH_ZOOM = 0.0005
+  const DETAIL_INCREASE_RATE = 1.5 // Higher = min delta lower at intermediate zoom levels = more detail at intermediate zoom levels
+
+  const minPosΔ = scalePow()
+    .clamp(true)
+    .exponent(DETAIL_INCREASE_RATE)
+    .range([MIN_POS_Δ_HIGH_ZOOM, MIN_POS_Δ_LOW_ZOOM])
+    .domain([0, 1])(normalizedZoom)
+
+  return minPosΔ
+}
+
 export interface TrackGeneratorConfig extends GeneratorConfig {
   data: FeatureCollection
   color?: string
 }
 
+const memoizedById: Dictionary<Dictionary<(...args: any[]) => any>> = {}
+const memoizeById = (id: string, fun: (...args: any[]) => any) => {
```

Comment:
This is good for now but as you said we'll have to improve as should be used to avoid this too https://github.com/GlobalFishingWatch/track-inspector/pull/19/files#diff-66dc89e18da30bfa63fcce67e60a1e7dR6